vk_video_decoder/libs/VkVideoDecoder/VkVideoDecoder.cpp

```diff
@@ -693,7 +686,7 @@ int VkVideoDecoder::DecodePictureWithParameters(VkParserPerFrameDecodeParameters
     VkVideoEndCodingInfoKHR decodeEndInfo = { VK_STRUCTURE_TYPE_VIDEO_END_CODING_INFO_KHR };
     m_vkDevCtx->CmdEndVideoCodingKHR(frameDataSlot.commandBuffer, &decodeEndInfo);
 
-    if (m_useSeparateOutputImages || m_useLinearOutput) {
+    if ((!m_useSeparateOutputImages) && m_useLinearOutput) {
         CopyOptimalToLinearImage(frameDataSlot.commandBuffer,
```

Comment:
if (m_useSeparateOutputImages || m_useLinearOutput)
This above condition should stay the way it is, but add a third one for the the OUTPUT_DISTINCT;
if (!outputDistinct && (m_useSeparateOutputImages || m_useLinearOutput)) {
...
}